src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -512,6 +602,27 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
     triggerAutomation(this.hass, (ev.target as any).stateObj.entity_id);
   }
 
+  private _preprocessYaml() {
+    const cleanConfig = deepClone(this._config);
+    delete cleanConfig.id;
+    return cleanConfig;
+  }
```

Comment:
For explanation of why this is a thing: We do not want the user to mess with the `id` property, and we don't want it to be copied and pasted either.

I plan more uses for this function.

Reply:
Actually, come to think of it those uses won't apply to scripts anyway. But maybe it should be left in there for consistency?

Reply:
Yes I need this for the grid layout as well. I just didn't know the best way to do it. Thanks! :)